src/nvim/decoration_provider.c

```diff
@@ -16,8 +16,15 @@ static kvec_t(DecorProvider) decor_providers = KV_INITIAL_VALUE;
     LUA_NOREF, LUA_NOREF, LUA_NOREF, \
     LUA_NOREF, -1 }
 
-static bool decor_provider_invoke(NS ns_id, const char *name, LuaRef ref, Array args,
-                                  bool default_true, char **perr)
+static void decor_provider_error(DecorProvider *provider, const char *name, Error err)
+{
+  const char *ns_name = describe_ns(provider->ns_id);
+  ELOG("error in provider %s.%s: %s", ns_name, name, err.msg);
+  semsg_multiline("Error in decoration provider %s.%s:\n%s", ns_name, name, err.msg);
```

Comment:
do we also disable the provider after one (or three, like timer) error messages? otherwise the editor might become unusable due to emsg flood.

Reply:
As we return false in `decoration_provider_invoke` I'd say that this is not necessary, but maybe we can.

And I'd go with only one error to disable, so that we keep the thing simple.

Reply:
Better idea. Why do we add an `on_error` callback and let providers decide on what they want to do?

Default to emsg despite how intrusive it might be.

Reply:
I'd rather not. This part of the code handles the errors hapenning _within_ a decoration provider, and should thus be handled in core.

Reply:
That reasoning isn't very clear. The callbacks are implemented by the providers, they have every right to be able to provide a handler for errors.